source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -81,6 +81,12 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
     service_name_matcher_.emplace(config.http_health_check().service_name_matcher());
   }
 
+  if (!Http::HeaderUtility::headerValueIsValid(path_) ||
+      !Http::HeaderUtility::headerValueIsValid(host_value_)) {
```

Comment:
For host consider using: `HeaderUtility::authorityIsValid`.

Reply:
Why? Please add a comment with the reason.

Reply:
The initial thought is to keep the same validation as the beginning of `encodeHeaders()`, but I don't have objections to separate them. Done  & PTAL